diff --git a/node_modules/react-native-iap/android/src/play/java/com/dooboolab/rniap/RNIapModule.kt b/node_modules/react-native-iap/android/src/play/java/com/dooboolab/rniap/RNIapModule.kt
index 54ca696..0cd5e72 100644
--- a/node_modules/react-native-iap/android/src/play/java/com/dooboolab/rniap/RNIapModule.kt
+++ b/node_modules/react-native-iap/android/src/play/java/com/dooboolab/rniap/RNIapModule.kt
@@ -13,6 +13,7 @@ import com.android.billingclient.api.ConsumeParams
 import com.android.billingclient.api.ConsumeResponseListener
 import com.android.billingclient.api.GetBillingConfigParams
 import com.android.billingclient.api.GetBillingConfigParams.Builder
+import com.android.billingclient.api.PendingPurchasesParams
 import com.android.billingclient.api.ProductDetails
 import com.android.billingclient.api.Purchase
 import com.android.billingclient.api.PurchaseHistoryRecord
@@ -42,7 +43,7 @@ import com.google.android.gms.common.GoogleApiAvailability
 @ReactModule(name = RNIapModule.TAG)
 class RNIapModule(
     private val reactContext: ReactApplicationContext,
-    private val builder: BillingClient.Builder = BillingClient.newBuilder(reactContext).enablePendingPurchases(),
+    private val builder: BillingClient.Builder = BillingClient.newBuilder(reactContext).enablePendingPurchases(PendingPurchasesParams.newBuilder().enableOneTimeProducts().build()),
     private val googleApiAvailability: GoogleApiAvailability = GoogleApiAvailability.getInstance(),
 ) : ReactContextBaseJavaModule(reactContext),
     PurchasesUpdatedListener {
@@ -461,7 +462,7 @@ class RNIapModule(
         isOfferPersonalized: Boolean, // New parameter in V5
         promise: Promise,
     ) {
-        val activity = currentActivity
+        val activity: android.app.Activity? = reactApplicationContext.currentActivity
         if (activity == null) {
             promise.safeReject(PromiseUtils.E_UNKNOWN, "getCurrentActivity returned null")
             return
